builder/dispatchers.go

```diff
@@ -253,6 +253,16 @@ func run(b *Builder, args []string, attributes map[string]bool, original string)
 	c.Mount()
 	defer c.Unmount()
 
+	// Unset the volumes
+	volumeConfig := c.Config.Volumes
+	c.Config.Volumes = nil
+	volumes := c.Volumes
+	c.Volumes = nil
+	defer func() {
+		c.Volumes = volumes
+		c.Config.Volumes = volumeConfig
```

Comment:
oh poo, yep. I moved this from somewhere else and never updated.
Fixed now and beefed up the test which didn't catch this before.